app/controllers/api/services_controller.rb

```diff
@@ -19,6 +27,61 @@ def reconfigure_resource(type, id = nil, data = nil)
 
     private
 
+    def build_service_attributes(data)
+      attributes                           = data.dup
+      attributes['orchestration_manager']  = fetch_ext_management_system(data['orchestration_manager']) if data['orchestration_manager']
+      attributes['configuration_manager']  = fetch_ext_management_system(data['configuration_manager']) if data['configuration_manager']
+      attributes['orchestration_template'] = fetch_orchestration_template(data['orchestration_template']) if data['orchestration_template']
+      attributes['job_template']           = fetch_configuration_script(data['job_template']) if data['job_template']
+      # A nasty bug in Rack, a query_parser's normalize_params method can handle only 100 keys total in payload data
+      # since it converts it all to array and process it recursively with max depth 100. For now, we can hide the
+      # data from Rack with to_yaml. But normalize_params needs to be rewritten to cycles instead of recursion.
+      attributes['job_options']            = YAML.load(data['job_options']) if data['job_options']
```

Comment:
@imtayadeway this is my hack around of the Rack bug we found. Not sure we can fix Rack before the release.